src/routes/firmware.ts

```diff
@@ -0,0 +1,82 @@
+import { spawn, execSync } from 'child_process';
+import { CAMERA_TYPE } from 'config';
+
+import { Router } from 'express';
+import { CameraType } from 'types';
+const router = Router();
+
+const HDC_ROOT = '/mnt/data/'
+const HDCS_ROOT = '/data/'
+const MENDER_PATH = HDCS_ROOT + 'core.mender'
+const FIP_PATH = HDCS_ROOT + 'fip.bin'
+
+let message = 'started';
+
+const runSpawn = (cmd: string) => {
+    const child = spawn(cmd, {
+        shell: true
+    });
+    child.stdout.setEncoding('utf8');
+    child.stdout.on('data', (data) => {
+        message = data.toString();
+    });
+
+    child.on('error', (error) => {
+        message = "Error" + error.toString();
+        console.error(`Error in Spawn process for cmd ${cmd} stderr: ${error}`);
+    });
+
+    child.on('close', (code, err) => {
+        if (code !== 0) {
+            message = "Closing spawn with code" + code?.toString() + "Error:" + err?.toString();
+            console.error(`Closing spawn due to error for cmd ${cmd} stderr: ${err?.toString()}`);
+        } else {
+            message = "Spawn ran successfully"
+        }
+    });
+};
+
+router.get('/install', async (req, res) => {
+
+    const firmwareFile = req?.body?.fileName || '';
+
+    try {
+        if (CAMERA_TYPE === CameraType.Hdc) {
+            try {
+                execSync(`test -f ${HDC_ROOT + firmwareFile}`, {
+                    encoding: 'utf-8',
+                });
+            } catch (error: unknown) {
+                console.log("Rauc file is not present")
+            }
+            runSpawn(`rauc install ${HDC_ROOT + firmwareFile} && reboot`)
```

Comment:
@HariHive @harroyoc 
Have we had a plan to leave the phone a chance to wait for installation to finish and sent "reboot" command?
Or you guys want the dashcam to fully own it?

There's a point of letting the dashcam restart itself, but I'm worried if the phone will be able to properly handle and receive the updated state in this case

Reply:
Discussed offline, we want app to handle reboot